jaxrs/src/main/java/com/facebook/airlift/jaxrs/thrift/ThriftMapper.java

```diff
@@ -72,7 +75,7 @@ public Object readFrom(
         }
         catch (Exception e) {
             // we want to return a 400 for bad Thrift but not for a real IO exception
-            if (e instanceof IOException && !(e instanceof EOFException)) {
+            if (e instanceof IOException && !(e instanceof ThriftProtocolException) && !(e instanceof EOFException)) {
```

Comment:
Should this be in a separate commit?  Doesn't seem related?

Reply:
Done. splitted into two commits